client/request/request.go

```diff
@@ -89,10 +95,7 @@ func (b *BearerAuthorizer) AuthHeader() string {
 // Req performs a request with the specified request options.
 func Req(opts *Options) (*http.Response, error) {
 	var scheme string
-	if opts.DisableSecure {
-		if !config.IsDevRelease() {
-			panic("Should not disable HTTPs")
-		}
+	if atomic.LoadInt32(&httpFallback) == 1 {
```

Comment:
This is just to be extra-careful to keep the `Req` method thread-safe, since the `httpFallback` var is global.